src/main.js

```diff
@@ -1,70 +1,81 @@
   handleCategoriesSelected();
   handleTypeSelector();
-  changeToolbarColor();
 }
 
+function rgbToHex (rgb) { 
+  var hex = Number(rgb).toString(16);
+  if (hex.length < 2) {
+       hex = "0" + hex;
+  }
+  return hex;
+};
+
 let neighbourhoodFilter = '';
 let roomTypeFilter = '';
+let ramp = '';
 
 function onCategoriesChanged (categories) {
   const categoriesWidget = document.getElementById('neighbourhoods');
   const data = categories.map((category) => {
+    // We need to map { x, y } (the format returned by VL) to { name, value } (the format needed by Airship)
     const { x, y } = category;
-    return {
+    let obj = {
       name: x,
       value: y
     };
+    if (ramp[obj.name]) {
+      obj.color = ramp[obj.name];
+    }
+    return obj;
   });
   categoriesWidget.categories = data;
 }
 
 function onAveragePriceChanged (averagePrice) {
   const formulaTextElement = document.getElementById('formula-number');
-  const formattedText = `${averagePrice.toFixed(2)}€`;
+  const formattedText = `${averagePrice.toFixed(2)}$`;
   formulaTextElement.textContent = formattedText;
 }
 
-// Event handling
-function handleCategoriesSelected () {
-  const categoriesWidget = document.getElementById('neighbourhoods');
-  categoriesWidget.addEventListener('categoriesSelected', (event) => {
-    filterNeighbourhood(event.detail);
+function onPriceChanged (price) {
+  var histogramWidget = document.getElementById('price');
+  const data = price.map(bin => {
+    // We need to map the format returned by VL to the format needed by Airship
+    return {
+      start: bin.x[0],
+      end: bin.x[1],
+      value: bin.y
+    };
   });
+  histogramWidget.data = data;
 }
 
-// Getting the filter
-function filterNeighbourhood (neighbourhoods) {
-  if (neighbourhoods.length > 0) {
-    const formattedData = neighbourhoods.map((neighbourhood) => {
-      return `'${neighbourhood}'`;
+function onLegendCalculated (legend) {
+  if (!ramp) {
+    ramp = {};
+    legend.data.forEach(entry => {
+      let hex = '#';
+      hex += rgbToHex(entry.value.r);
+      hex += rgbToHex(entry.value.g);
+      hex += rgbToHex(entry.value.b);
+
+      ramp[entry.key] = hex;
     });
-    neighbourhoodFilter = `$neighbourhood_group in [${formattedData.join(',')}]`;
-  } else {
-    neighbourhoodFilter = '';
   }
-  combineFilters(neighbourhoodFilter);
 }
 
-// Applying filter to viz
-function applyFilter(filter) {
-  const color = 'rgb(0,0,255)';
-  const opacity = '0.25';
-  let colorExp = `opacity(${color}, ${opacity})`;
-
-  if (!viz) {
-    return;
-  }
-
-  if (filter) {
-    colorExp = `opacity(${color}, ${filter} * ${opacity})`;
-  }
-  viz.color.blendTo(colorExp);
+function handleCategoriesSelected () {
+  const categoriesWidget = document.getElementById('neighbourhoods');
+  categoriesWidget.addEventListener('categoriesSelected', async (event) => {
+    const selected = await categoriesWidget.getSelectedCategories();
```

Comment:
This is going to be fun to explain (the async/await) :grimacing: 

Although, I think promises would be even worse.